test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java

```diff
@@ -0,0 +1,334 @@
+				{ "15.10.0.0.0.12.27",  "10.0.0.0" },
+				{ "100.10.0.0.0.10.12", "10.0.0.0" }, 
+				{ "255.10.0.0.0:6555",  "10.0.0.0:6555" },
+
+				{ "10.0.0.0:128",       "10.0.0.0:128" },
+				{ "ip-10.0.0.0",        "10.0.0.0" },
+				{ "IP-10.0.0.0",        "10.0.0.0" }
+		};
+		for (int i = 0; i < data.length; i++) {
+			String candidate = data[i][0];
+			String evidence = data[i][1];
+			HttpMessage msg = createHttpMessage(candidate);
+			rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+			assertThat(alertsRaised.size(), equalTo(i + 1));
+			assertThat(alertsRaised.get(i).getEvidence(), equalTo(evidence));
+			validateAlert(alertsRaised.get(i));
+		}
+	}
+
+	@Test
+	public void passesIfNonPrivateOrInvalidIp() throws HttpMalformedHeaderException {
+		String[] candidates = new String[] {
+				// the "borders" of private IP ranges
+				"9.255.255.255",
+				"11.0.0.0",
+				"172.15.255.255",
+				"172.32.0.0",
+				"192.167.255.255",
+				"192.169.0.0",
+				// outside & between the private ranges
+				"8.8.8.8",
+				"26.10.3.11",
+				"84.168.27.12",
+				"127.0.0.1",
+				"186.27.16.2",
+				"255.255.255.255",
+				// some invalid ones
+				"10",
+				"10.0.0",
+				"10.0.0:0",
+				"10/0.0.0",
+				"10.0\n0.0",
+				"10.0.0 0",
+				"10.0.0.a",
+				"999.0.0.0", 
+				"10.999.0.0",
+				"10.0.756.0",
+				"ip-10-0-0-256",
+				// no word boundaries
+				"2050:10.0.0.0bla",
+				"205010.0.0.0:bla",
+				"abcd10.0.0.0bla",
+				"abcd10.0.0.999",
+				"ip-10.0.0.9999",
+		};
+		for (String candidate : candidates) {
+			HttpMessage msg = createHttpMessage(candidate);
+			rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		}
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void alertsIfPrivIpAndAddsPortToEvidence() throws HttpMalformedHeaderException {
+		// ip and aws-hostname which get caoncatenated with the ports as candidates
+		String[] ipHost = new String[] {"10.0.0.0", "ip-10-0-0-0"};
+		// several ports in the range 0-65535
+		String[] ports = new String[] {
+				":0",
+				":1",     ":6",     ":9",
+				":10",    ":25",    ":99",
+				":100",   ":443",   ":999",
+				":1000",  ":8080",  ":9999",
+				":10000", ":16443", ":19999",
+				":20000", ":24128", ":29999",
+				":30000", ":35097", ":39999",
+				":40000", ":41962", ":49999",
+				":50000", ":56481", ":59999",
+				":61000", ":61443", ":61999",
+				":62000", ":62128", ":62999",
+				":63000", ":63097", ":63999",
+				":64000", ":64962", ":64999",
+				":65000", ":65010", ":65029",
+				":65100", ":65210", ":65329",
+				":65400", ":65410", ":65499",
+				":65500", ":65510", ":65518",
+				":65520", ":65529", ":65535",
+		};
+		for (int ci = 0; ci < ports.length; ci++) {
+			for (int ii = 0; ii < ipHost.length; ii++) {
+				String candidate = ipHost[ii] + ports[ci];
+				HttpMessage msg = createHttpMessage(candidate);
+				rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+				
+				int alertNoIdx = ci * 2 + ii;
```

Comment:
The list of alerts raised could be cleared prior scanning, reduces the memory used and the index would be always the same.